### PR TITLE
Update Dockerfile.okd and add OKD package list

### PR DESCRIPTION
**Pull request by LorbusChris**
- Updates builder image to Fedora 34
- Updates base image to CentOS 8 Stream
- Updates the link to the script for creating the
OpenStack yum repo files
- Syncs with updates made to Dockerfile.ocp
- Uses an OKD-specific main-packages-list

### CODE REVIEW

**Comment by dtantsur** [comment]
I think you need to change https://github.com/openshift/ironic-image/blob/master/ironic-config/ironic.conf.j2#L49 to `centos` to match what IPA will use.

File: Dockerfile.okd [hunk 1]
```diff
@@ -16,47 +16,49 @@ RUN if [ $(uname -m) = "x86_64" ]; then \
...
 COPY ${PKGS_LIST} ${EXTRA_PKGS_LIST:-$PKGS_LIST} /tmp/
-COPY prepare-image.sh /bin/
+COPY prepare-image.sh prepare-ipxe.sh /bin/
 
-RUN dnf install -y python3 python3-requests && \
-    curl https://raw.githubusercontent.com/openstack/tripleo-repos/master/tripleo_repos/main.py | python3 - -b master current-tripleo && \
-    prepare-image.sh && \
-    rm -f /bin/prepare-image.sh
+# Configure OpenStack repos
+RUN dnf install --setopt=install_weak_deps=False --setopt=tsflags=nodocs -y python3 && \
```
Comment (dtantsur) [inline Dockerfile.okd:28]: You need to install python3-requests, otherwise build fails.

**Comment by LorbusChris** [comment]
> I think you need to change https://github.com/openshift/ironic-image/blob/master/ironic-config/ironic.conf.j2#L49 to `centos` to match what IPA will use.

@dtantsur are you sure? It looks to me like this is the image built in the build stage of the Dockerfile.okd in this repo (which is Fedora-based)

**Comment by dtantsur** [comment]
> It looks to me like this is the image built in the build stage of the Dockerfile.okd in this repo (which is Fedora-based: https://github.com/openshift/ironic-image/blob/master/Dockerfile.okd#L6-L14)

Ugh. Yeah, you're probably right. We should eventually migrate to CentOS everywhere to avoid this sort of confusion.

**Comment by LorbusChris** [comment]
@dtantsur mind taking another look?

File: main-packages-list.okd [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+httpd
+httpd-tools
+ipmitool
+iproute
+ipxe-bootimgs
+ipxe-roms-qemu
+iscsi-initiator-utils
+mariadb-server
+mod_ssl
+openstack-ironic-api >= 1:18.1.1-0.20210812092216.4aec741.el8
```
[the remark this answers is not in the document]
Comment (LorbusChris) [reply main-packages-list.okd:15]: I've tested building the image locally, and it did work. I can certainly remove the versions though if that's preferred

[the remark this answers is not in the document]
Comment (elfosardo) [reply main-packages-list.okd:15]: if it builds it's fine, in the end it's up to you, it's just easier to maintain without specific versions, also consider that the versions mentioned here are related to RedHat built packages and differ from the RDO ones.
If you need minimum versions I suggest to check what's in the tripleo repos used by the OKD build, but to be honest I would just use the file at https://github.com/metal3-io/ironic-image/blob/master/main-packages-list.txt without versions.

[the remark this answers is not in the document]
Comment (LorbusChris) [reply main-packages-list.okd:15]: Makes sense. I'll remove the versions upon rebasing after #208 lands.

**Comment by elfosardo** [comment]
@LorbusChris #208 has merged, can you please rebase ?

**Comment by LorbusChris** [comment]
@elfosardo, updated :)

**Comment by elfosardo** [comment]
/lgtm
was approved already! :)

**Pull request merged**